.github/workflows/ci.yml

```diff
@@ -60,6 +60,15 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: ${{ env.go-version }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: cache-go-${{ env.go-version }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            cache-go-${{ env.go-version }}-
+            cache-
```

Comment:
This is a bit confusing to me; we want to restore a cache built for that exact go version and the exact dependencies. If we didn't find one, then we restore the next best built for the same go version. That's OK so far, but if we don't find one for that go version this action restores any cache?

I think just `cache-go-${{ env.go-version }}-` is enough as a fallback and we just rebuild our caches when the go version changes.